File: data/src/main/java/com/tonitealive/app/data/DefaultMessageProvider.java
package com.tonitealive.app.data;

import android.content.Context;

import com.tonitealive.app.domain.MessageId;
import com.tonitealive.app.domain.MessageProvider;

import java.util.HashMap;
import java.util.Map;

public class DefaultMessageProvider implements MessageProvider {

    private final Context context;
    private final Map<MessageId, Integer> messageCodes = new HashMap<>();

    public DefaultMessageProvider(Context context) {
        this.context = context;
        initMessageCodes();
    }

    @Override
    public String getMessage(MessageId id) {
        Integer code = messageCodes.get(id);
        if (code != null)
            return context.getString(code);
        throw new IllegalArgumentException("No messsage found for id " + id);
    }

    private void initMessageCodes() {
        messageCodes.put(MessageId.SIGN_IN_SUCCESS, R.string.signin_succeeded);
    }
}

File: data/src/main/kotlin/com/tonitealive/app/data/DefaultMessageProvider.kt
package com.tonitealive.app.data

import android.content.Context
import com.tonitealive.app.domain.MessageId
import com.tonitealive.app.domain.MessageProvider
import java.util.*
import javax.inject.Singleton

@Singleton
class DefaultMessageProvider(private val context: Context) : MessageProvider {

    private val messageCodes: HashMap<MessageId, Int> = HashMap()

    init {
        messageCodes.put(MessageId.SIGN_IN_SUCCESS, R.string.signin_succeeded)
    }

    override fun getMessage(id: MessageId): String {
        val code = messageCodes[id]
        if (code != null)
            return context.getString(code)
        throw IllegalArgumentException("No message found for id $id")
    }
}